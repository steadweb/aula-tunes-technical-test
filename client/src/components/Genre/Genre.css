.selected-genre {
  padding: 20px;
  border: 1px solid var(--secondary-color);
}

.genre-container {
  display: flex;
  justify-content: center;
  margin: 20px auto;
}

.pill {
  border: 1px solid var(--secondary-color);
  background-color: var(--secondary-color);
  color: black;
  padding: 12px;
  border-radius: 6px;
  cursor: pointer;
  transition: background-color 0.3s;
}

.pill:hover {
  background-color: var(--primary-color);
  color: var(--secondary-color);
}

.pill::after {
  content: '\2716';
  margin: 6px;
}
