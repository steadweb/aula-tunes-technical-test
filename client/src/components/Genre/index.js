import React from 'react';
import './Genre.css';

function GenreContainer({ genre, setGenre }) {
  return (
    <>
      {genre && (
        <div className='genre-container'>
          <div className='pill' onClick={() => setGenre(null)}>
            {genre.name}
          </div>
        </div>
      )}
    </>
  );
}

export default GenreContainer;
