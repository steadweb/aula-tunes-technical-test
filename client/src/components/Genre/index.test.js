import React from 'react';
import renderer from 'react-test-renderer';
import Genre from './index';

describe('Genre', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Genre />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
