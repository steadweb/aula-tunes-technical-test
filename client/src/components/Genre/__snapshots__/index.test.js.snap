// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre renders correctly 1`] = `null`;
